src/client/Client.cc

```diff
@@ -11114,6 +11114,11 @@ int64_t Client::_read(Fh *f, int64_t offset, uint64_t size, bufferlist *bl,
 {
   ceph_assert(ceph_mutex_is_locked_by_me(client_lock));
 
+  // zero bytes read is not supported by osd
+  if (size < 1) {
```

Comment:
If we are looking only for 0, I suggest something like `size == 0` That way seems more straight forward.